packages/react-compass/src/popover/popover.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {useCallback} from 'react'
+import {StyledComponentProps} from '../utils/stitches.types'
+import {useDOMRef} from '../utils/use-dom-ref'
+import {StyledPopover, StyledPopoverContent} from './popover.style'
+
+export type PopoverDirection =
+  | 'bottom-right'
+  | 'bottom-left'
+  | 'top-right'
+  | 'top-left'
+
+interface Props extends StyledComponentProps {
+  children?: React.ReactNode
+  direction?: PopoverDirection
+  isOpen: boolean
+  anchor: React.ReactNode
+  onClose?: () => void
+}
+
+export type PopoverProps = Props &
+  Omit<React.HTMLAttributes<HTMLDivElement>, keyof Props>
+
+const Popover = React.forwardRef<HTMLDivElement, PopoverProps>((props, ref) => {
+  const {
+    children,
+    anchor,
+    isOpen,
+    css = {},
+    direction = 'bottom-right',
+    onClose = () => {},
+    ...delegated
+  } = props
+
+  const popoverRef = useDOMRef<HTMLDivElement>(ref)
+
+  const handlePopoverBlur = () => {
+    onClose()
+  }
+
+  const handleWrapperBlur = useCallback(() => {
+    requestAnimationFrame(() => {
+      if (
+        popoverRef.current &&
+        !popoverRef.current.contains(document.activeElement)
+      ) {
+        handlePopoverBlur()
+      }
+    })
+  }, [popoverRef, handlePopoverBlur])
+
+  return (
+    <StyledPopover
+      className='cdg-popover'
+      ref={popoverRef}
+      css={css}
+      {...delegated}
+      onBlur={handleWrapperBlur}
+      tabIndex={0}
+    >
+      {anchor}
+      {isOpen && (
+        <StyledPopoverContent className={`cdg-popover-content ${direction}`}>
```

Comment:
I think need to handle auto display position. For example, I use popover component and passed with the `top-right` position to prop, but when the button is at the top of the page, it should automatically display in the bottom-right position.

Demo: https://react-spectrum.adobe.com/react-spectrum/Menu.html

Reply:
Hi @thaycacac ,

Thanks for your advice here!
I think our components are using popover, they are on a fixed position. However, to handle the position dynamically, we will need to spend a lot of effort. So I would not support it for now.
Could you please help to convert it to a ticket then we can support more features for Popover later?

Reply:
ok. thanks